torch/distributed/_tensor/ops/math_ops.py

```diff
@@ -104,7 +106,10 @@ def map_placements_after_reduction(
             if new_shard_dim == -1 or shard_dim in reduction_dims:
                 # if new_shard_dim collapsed or its in the reduction dims
                 # (i.e. for the case where keepdims=True), we generate partial
-                new_placements.append(_Partial(reduction_op))
+                if isinstance(reduction_op, _Partial):  # HACK!
```

Comment:
This is a bad hack, but I was not sure an elegant way to specify a custom `_Partial` placement here. This would need to be fixed if we want to land in the future.